conan/tools/meson/meson.py

```diff
@@ -159,21 +179,95 @@ def _native_content(self):
+                      'Linux': 'linux',
+                      'SunOS': 'sunos',
+                      'Windows': 'windows',
+                      'WindowsCE': 'windows',
+                      'WindowsStore': 'windows'}
+        # https://mesonbuild.com/Reference-tables.html#cpu-families
+        cpu_family_map = {'armv4': ('arm', 'armv4', 'little'),
+                          'armv4i': ('arm', 'armv4i', 'little'),
+                          'armv5el': ('arm', 'armv5el', 'little'),
+                          'armv5hf': ('arm', 'armv5hf', 'little'),
+                          'armv6': ('arm', 'armv6', 'little'),
+                          'armv7': ('arm', 'armv7', 'little'),
+                          'armv7hf': ('arm', 'armv7hf', 'little'),
+                          'armv7s': ('arm', 'armv7s', 'little'),
+                          'armv7k':('arm', 'armv7k', 'little'),
+                          'armv8': ('aarch64', 'armv8', 'little'),
+                          'armv8_32': ('aarch64', 'armv8_32', 'little'),
+                          'armv8.3': ('aarch64', 'armv8.3', 'little'),
+                          'avr': ('avr', 'avr', 'little'),
+                          'mips': ('mips', 'mips', 'big'),
+                          'mips64': ('mips64', 'mips64', 'big'),
+                          'ppc32be': ('ppc', 'ppc', 'big'),
+                          'ppc32': ('ppc', 'ppc', 'little'),
+                          'ppc64le': ('ppc64', 'ppc64', 'little'),
+                          'ppc64': ('ppc64', 'ppc64', 'big'),
+                          's390': ('s390', 's390', 'big'),
+                          's390x': ('s390x', 's390x', 'big'),
+                          'sh4le': ('sh4', 'sh4', 'little'),
+                          'sparc': ('sparc', 'sparc', 'big'),
+                          'sparcv9': ('sparc64', 'sparc64', 'big'),
+                          'wasm': ('wasm32', 'wasm32', 'little'),
+                          'x86': ('x86', 'x86', 'little'),
+                          'x86_64': ('x86_64', 'x86_64', 'little')}
+        system = system_map.get(machine_os, machine_os.lower())
+        default_cpu_tuple = (machine_arch.lower(), machine_arch.lower(), 'little')
+        cpu_tuple = cpu_family_map.get(machine_arch, default_cpu_tuple)
+        cpu_family, cpu, endian = cpu_tuple[0], cpu_tuple[1], cpu_tuple[2]
+        context = {
+            'system': self._to_meson_value(system),
+            'cpu_family': self._to_meson_value(cpu_family),
+            'cpu': self._to_meson_value(cpu),
+            'endian': self._to_meson_value(endian),
+        }
+        t = Template(self._machine_template)
+        return t.render(context)
 
     @property
     def _cross_content(self):
-        raise Exception("cross-building is not implemented yet!")
+        os_build, arch_build, os_host, arch_host = get_cross_building_settings(self._conanfile)
+        os_target, arch_target = os_host, arch_host  # assume target the same as a host for now?
```

Comment:
If we can ignore the `target_machine` maybe it is better (Conan doesn't implement it), if not, at least add a TODO/FIXME

```suggestion
        os_target, arch_target = os_host, arch_host  # TODO: assume target the same as a host for now?
```